phpBB/includes/functions.php

```diff
@@ -921,6 +921,16 @@ function markread($mode, $forum_id = false, $topic_id = false, $post_time = 0, $
 
 		return;
 	}
+	else if ($mode == 'usermention')
```

Comment:
It only needs to call the markread when the post is marked as unread before, otherwise you won't have a notification if it anyway. This is how the quote notification works, and we should do the same with the mention notification